### PR TITLE
feature/UWEB-6230-filter-update

### PR DESCRIPTION
**Pull request by dan-entous**

### CODE REVIEW

File: src/js/components/atoms/label-filter/label-filter.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /** @format */
 
-import { Vue, Component, Prop } from 'vue-property-decorator';
+import { LabelFilterInterface } from 'components/atoms/label-filter/label-filter.interface';
+import { Vue, Component, Prop, Watch } from 'vue-property-decorator';
```
Comment (matijs) [inline src/js/components/atoms/label-filter/label-filter.ts]: `Watch` isn't used, doesn't ESLint complain?

File: src/js/components/atoms/label-filter/label-filter.scss [hunk 2]
```diff
@@ -1,41 +1,45 @@
...
+    border: 1px solid $colorGrayDark;
+    color: $colorBlack;
+    cursor: pointer;
+    display: inline-block;
+    font-size: 12px;
+    font-weight: 500;
+    padding: 0 $buildingUnit / 2;
+}
+
+.label-filter > :checked:not(:disabled) + .label-filter__label {
```
Comment (matijs) [inline src/js/components/atoms/label-filter/label-filter.scss]: I don't think `:not()f` is needed here.

File: src/js/components/atoms/label-filter/label-filter.scss [hunk 3]
```diff
@@ -1,41 +1,45 @@
...
-.label-filter__label {
-    display: inline-block;
-
-    padding-left: $buildingUnit / 2;
-    padding-right: $buildingUnit / 2;
-
-    font-size: 12px;
-    font-weight: 500;
-
+.label-filter > :disabled:not(:checked) + .label-filter__label {
```
Comment (matijs) [inline src/js/components/atoms/label-filter/label-filter.scss]: Same, `:not()` is not needed, this is just the `:disabled` state.

File: src/js/components/atoms/label-filter/label-filter.scss [hunk 4]
```diff
@@ -1,41 +1,45 @@
...
+    background-color: $colorWhite;
+    border-color: $colorGrayLight;
     color: $colorGrayDark;
-    background-color: $colorGrayExtraLight;
+    cursor: not-allowed;
+}
 
-    border-radius: 2px;
-    border: 1px solid $colorGray;
+.label-filter > :disabled:checked + .label-filter__label {
```
Comment (matijs) [inline src/js/components/atoms/label-filter/label-filter.scss]: This is the `:disabled:checked` state that overwrites the two above as long as the order in this file is correct.

File: src/js/components/atoms/label-filter/label-filter.ts [hunk 5]
```diff
@@ -12,23 +13,22 @@ import { LabelFilterProps } from './label-filter.models';
...
     }
 
     onChange($event) {
-        const payload = {
-            ...this.payload,
-            checked: $event.target.checked,
-        };
-
-        this.$emit('label-filter-change', payload);
+        this.emitEvent(this.eventChange);
```
Comment (matijs) [inline src/js/components/atoms/label-filter/label-filter.ts]: Left over from when this was more complicated, `emitEvent` does not need to be its own method anymore.

File: src/js/components/molecules/filter-category/filter-category.ts [hunk 6]
```diff
@@ -1,48 +1,81 @@
...
 import { FilterCategoryProps } from './filter-category.models';
+import { LabelFilterInterface } from 'components/atoms/label-filter/label-filter.interface';
 
 @WithRender
-@Component({
-    name: 'FilterCategory',
-})
+@Component({ name: 'FilterCategory' })
 export class FilterCategory extends Vue implements FilterCategoryProps {
+    @Prop({ type: Array }) activeFilters?: FilterCategoryProps['activeFilters'];
```
Comment (matijs) [inline src/js/components/molecules/filter-category/filter-category.ts]:
```suggestion
    @Prop({ type: Array, default: [] }) activeFilters?: FilterCategoryProps['activeFilters'];
```
Maybe?

File: src/js/components/atoms/label-filter/label-filter.scss (hunk 2 above)
Comment (dan-entous) [reply src/js/components/atoms/label-filter/label-filter.scss]: want it to be specific so there's no need for css overwrite

File: src/js/components/atoms/label-filter/label-filter.scss (hunk 3 above)
Comment (dan-entous) [reply src/js/components/atoms/label-filter/label-filter.scss]: want it to be specific so there's no need for css overwrite

File: src/js/components/atoms/label-filter/label-filter.scss (hunk 4 above)
Comment (dan-entous) [reply src/js/components/atoms/label-filter/label-filter.scss]: want it to be specific so there's no need for css overwrite

File: src/js/components/atoms/label-filter/label-filter.ts (hunk 5 above)
Comment (dan-entous) [reply src/js/components/atoms/label-filter/label-filter.ts]: done

File: src/js/components/molecules/filter-category/filter-category.ts (hunk 6 above)
Comment (dan-entous) [reply src/js/components/molecules/filter-category/filter-category.ts]: done.

**Pull request merged**